sdks/python/apache_beam/runners/portability/spark_runner_test.py

```diff
@@ -21,141 +21,158 @@
-
-    def test_sdf_with_watermark_tracking(self):
-      # Skip until Spark runner supports SDF.
-      raise unittest.SkipTest("BEAM-7222")
-
-    def test_sdf_with_sdf_initiated_checkpointing(self):
-      # Skip until Spark runner supports SDF.
-      raise unittest.SkipTest("BEAM-7222")
-
-    def test_sdf_synthetic_source(self):
-      # Skip until Spark runner supports SDF.
-      raise unittest.SkipTest("BEAM-7222")
-
-    def test_external_transforms(self):
-      # Skip until Spark runner supports external transforms.
-      raise unittest.SkipTest("BEAM-7232")
-
-    def test_callbacks_with_exception(self):
-      # Skip until Spark runner supports bundle finalization.
-      raise unittest.SkipTest("BEAM-7233")
-
-    def test_register_finalizations(self):
-      # Skip until Spark runner supports bundle finalization.
-      raise unittest.SkipTest("BEAM-7233")
-
-    def test_flattened_side_input(self):
-      # Blocked on support for transcoding
-      # https://jira.apache.org/jira/browse/BEAM-7236
-      super(SparkRunnerTest,
-            self).test_flattened_side_input(with_transcoding=False)
-
-    # Inherits all other tests from PortableRunnerTest.
+# Run as
+#
+# pytest spark_runner_test.py \
+#     [--test_pipeline_options "--spark_job_server_jar=/path/to/job_server.jar \
+#                               --environment_type=DOCKER"] \
+#     [SparkRunnerTest.test_method, ...]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SparkRunnerTest(portable_runner_test.PortableRunnerTest):
+  _use_grpc = True
+  _use_subprocesses = True
+
+  expansion_port = None
+  spark_job_server_jar = None
+
+  @pytest.fixture(autouse=True)
+  def parse_options(self, request):
+    test_pipeline_options = (
+        request.config.option.test_pipeline_options
+        if request.config.option.test_pipeline_options else '')
+    parser = argparse.ArgumentParser(add_help=True)
+    parser.add_argument(
+        '--spark_job_server_jar',
+        help='Job server jar to submit jobs.',
+        action='store')
+    parser.add_argument(
+        '--environment_type',
+        default='LOOPBACK',
+        choices=['DOCKER', 'PROCESS', 'LOOPBACK'],
+        help='Set the environment type for running user code. DOCKER runs '
+        'user code in a container. PROCESS runs user code in '
+        'automatically started processes. LOOPBACK runs user code on '
+        'the same process that originally submitted the job.')
+    parser.add_argument(
+        '--environment_config',
+        help='Set environment configuration for running the user code.\n For '
+        'DOCKER: Url for the docker image.\n For PROCESS: json of the '
+        'form {"os": "<OS>", "arch": "<ARCHITECTURE>", "command": '
+        '"<process to execute>", "env":{"<Environment variables 1>": '
+        '"<ENV_VAL>"} }. All fields in the json are optional except '
+        'command.')
+    known_args, unknown_args = parser.parse_known_args(
+        test_pipeline_options.split())
```

Comment:
Same here, what about whitespace in the values?

Reply:
done